pkg/controller/volume/attach_detach_controller.go

```diff
@@ -456,28 +457,26 @@ func (adc *attachDetachController) getPVSpecFromCache(
 	return volume.NewSpecFromPersistentVolume(&clonedPV, pvcReadOnly), nil
 }
 
-// processSafeToDetachAnnotations processes the "safe to detach" annotations for
-// the given node. It makes calls to delete any annotations referring to volumes
-// it is not aware of. For volumes it is aware of, it marks them safe to detach
-// in the "actual state of world" data structure.
-func (adc *attachDetachController) processSafeToDetachAnnotations(
-	nodeName string, annotations map[string]string) {
-	var annotationsToRemove []string
-	for annotation := range annotations {
-		// Check annotations for "safe to detach" volumes
-		annotation = strings.ToLower(annotation)
-		if strings.HasPrefix(annotation, SafeToDetachAnnotation) {
-			// If volume exists in "actual state of world" mark it as safe to detach
-			safeToAttachVolume := strings.TrimPrefix(annotation, SafeToDetachAnnotation)
-			if err := adc.actualStateOfWorld.MarkVolumeNodeSafeToDetach(safeToAttachVolume, nodeName); err != nil {
-				// If volume doesn't exist in "actual state of world" remove
-				// the "safe to detach" annotation from the node
-				annotationsToRemove = append(annotationsToRemove, annotation)
+func (adc *attachDetachController) processVolumesInUse(
+	nodeName string, volumesInUse []api.UniqueDeviceName) {
+	for _, attachedVolume := range adc.actualStateOfWorld.GetAttachedVolumes() {
```

Comment:
Seems like it may be more efficient to add a method to `ActualStateOfTheWorld` called `GetAttachedVolumesForNode` or something that does the filtering in `asw`, since we're iterating through the entire collection of volumes again here.